phoenix-core/src/main/java/org/apache/phoenix/expression/ComparisonExpression.java

```diff
@@ -128,6 +131,27 @@ public static Expression create(CompareOp op, List<Expression> children, Immutab
             if ( ! ( lhsExpr instanceof RowValueConstructorExpression ) ) {
                 lhsExpr = new RowValueConstructorExpression(Collections.singletonList(lhsExpr), lhsExpr.isStateless());
             }
+
+            //At this point both sides should be in the same row format
+            //We add the inverts so the filtering can be done properly for mixed sort type RVCs,  The entire RVC has to
+            // be in ASC for the actual compare to work since compare simply does a varbyte compare.  See PHOENIX-4841
+            List<RowValueConstructorExpression> rvcList = Lists.newArrayList((RowValueConstructorExpression)lhsExpr,(RowValueConstructorExpression)rhsExpr);
```

Comment:
can this be a separate method? Then add a unit test to this?

Reply:
Ok